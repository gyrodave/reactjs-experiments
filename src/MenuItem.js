import React from 'react';
import 'babel-polyfill';
import { Button } from 'react-bootstrap';
import PropTypes from 'prop-types';

const MenuItem = ({ clickHandler, label }) => (
  <Button bsStyle="primary" onClick={clickHandler}>{label}</Button>
);

MenuItem.propTypes = {
  'clickHandler': PropTypes.func.isRequired,
  'label' : PropTypes.string.isRequired,
}

export default MenuItem;
