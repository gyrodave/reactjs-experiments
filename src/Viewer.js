import React from 'react';
import injectSheet from 'react-jss'

const styles = {
  viewer: {
    left: '0',
    'text-align': 'left'
  },
};

const Viewer = ({classes, text }) => {
  return (
    <div>
      {text.map(e => <div className={classes.viewer} key={e}>{e}</div>)}
    </div>
  );
};

export default injectSheet(styles)(Viewer);
