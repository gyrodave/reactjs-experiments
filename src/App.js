import React from 'react';
import './App.css';
import MenuItem from './MenuItem.js';
import Viewer from './Viewer.js';
import axios from 'axios';

const ELEM_EMPTY = 0;
const ELEM_CHARACTERS = 1;
const ELEM_TIMEBASED = 2;

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      element: 0,
      text: [],
    };
  }

  async clickHandler(action) {
    let url = '/api/';
    const response = [];

    switch (action) {
      case ELEM_CHARACTERS:
        url += 'characters';
        break;
      default:
        url += 'timebased';
        break;
    }

    const resp = await axios({
      method: 'get',
      url: url,
      responseType: 'application/json',
    });

    switch (action) {
      case ELEM_CHARACTERS:
        resp.data.characters.forEach(elem => {
          response.push(elem);
        });
        break;
      default:
        resp.data.elements.forEach(elem => {
          response.push(elem.id + ' : ' + elem.title);
        });
        break;
    }

    let newState = {
      ...this.state,
      element: action,
      text: response,
    }; // // const a = { //   ...this.state, //   element: action, // }; // // const b = { //   element: action, //   ...this.state, // }; // // console.log(a.element); // console.log(b.element);

    // console.log(`DÃ©part : ${this.state.element}`)
    this.setState(newState);
  }

  render() {
    return (
      <div className="App">
        <div id="menu">
          <MenuItem
            label="Characters"
            clickHandler={() => this.clickHandler(ELEM_CHARACTERS)}
            action={ELEM_CHARACTERS}
          />
          <MenuItem
            label="Timebased"
            clickHandler={() => this.clickHandler(ELEM_TIMEBASED)}
            action={ELEM_TIMEBASED}
          />
        </div>
        <Viewer text={this.state.text} />

      </div>
    );
  }
}
export default App;
