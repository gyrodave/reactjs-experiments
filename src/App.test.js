import React from 'react';
import {App, ELEM_EMPTY, ELEM_CHARACTERS, ELEM_TIMEBASED} from './App';
import MenuItem from './MenuItem';
import Viewer from './Viewer';
import { Button } from 'react-bootstrap';
import { shallow } from 'enzyme';
import MockAdapter from 'axios-mock-adapter';
import axios from 'axios';

var mock = new MockAdapter(axios);

it('test App components', () => {
  const wrapper = shallow(<App />);
  expect(wrapper.find(MenuItem).length).toBe(2);
  expect(wrapper.find(Viewer).length).toBe(1);
});
test('setState after clickHandler', async () => {
  const elementsTimebased = [];
  const respCharaters = {
    error: false,
    characters: [
      'Luke Skywalker',
      'C-3PO',
      'R2-D2'
    ],
    length: 3,
  };
  const respTimebased = {
    "length": 2,
    "elements": [
      {
        "id": "656xwp",
        "title": "Lmao someone made cod:mw3 into a real thing"
      },
      {
        "id": "656xv3",
        "title": "The compulsive gum swallower"
      }
    ]
  }

  respTimebased.elements.forEach( (elem) => {
    elementsTimebased.push(elem.id + ' : ' + elem.title);
  })

  mock.onGet('http://test:3000/api/characters').reply(200, respCharaters);
  mock.onGet('http://test:3000/api/timebased').reply(200, respTimebased);
  try {
    const wrapper = shallow(<App />);
    expect(wrapper.state('element')).toBe(ELEM_EMPTY);

    await wrapper.instance().clickHandler(ELEM_CHARACTERS);
    expect(wrapper.state('element')).toBe(ELEM_CHARACTERS);
    expect(wrapper.state('text')).toEqual(respCharaters.characters);

    await wrapper.instance().clickHandler(ELEM_TIMEBASED);
    expect(wrapper.state('element')).toBe(ELEM_TIMEBASED);
    expect(wrapper.state('text')).toEqual(elementsTimebased);
  } catch (err) {
    console.log(err);
  }
});
