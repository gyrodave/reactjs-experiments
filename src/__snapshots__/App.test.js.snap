// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`setState after clickHandler 1`] = `
<div
  className="App"
>
  <div
    id="menu"
  >
    <MenuItem
      clickHandler={[Function]}
      label="Characters"
    />
    <MenuItem
      clickHandler={[Function]}
      label="Timebased"
    />
  </div>
  <Jss(Viewer)
    text={Array []}
  />
</div>
`;

exports[`setState after clickHandler 2`] = `
<div
  className="App"
>
  <div
    id="menu"
  >
    <MenuItem
      clickHandler={[Function]}
      label="Characters"
    />
    <MenuItem
      clickHandler={[Function]}
      label="Timebased"
    />
  </div>
  <Jss(Viewer)
    text={Array []}
  />
</div>
`;

exports[`test App components 1`] = `
<div
  className="App"
>
  <div
    id="menu"
  >
    <MenuItem
      clickHandler={[Function]}
      label="Characters"
    />
    <MenuItem
      clickHandler={[Function]}
      label="Timebased"
    />
  </div>
  <Jss(Viewer)
    text={Array []}
  />
</div>
`;
